text/0000-cfg-path-version.md

```diff
@@ -508,22 +508,27 @@ with `GHC` and alternative Haskell compilers.
 
 ### The bikeshed - Argument syntax
 
-We have two options with respect to how the `version` flag may be specified:
+We have roughly two options with respect to how the `version` flag may be specified:
 
 1. `version = "<semver>"`
-2. `version("<semver>")`
+2. `version(<semver>)`
 
 The syntax in 2. is currently an error in `#[cfg(..)]` as you may witness with:
 
 ```rust
 // error[E0565]: unsupported literal
-#[cfg(abracadabra("1.27"))] fn bar() {}
+#[cfg(abracadabra(1.27))] fn bar() {}
+                  ^^^^
 ```
 
-We could allow this syntax. However, we have chosen the syntax in 1.
-This with consistency with flags such as `target_feature = "bmi"`.
-Another reason to go with `version = ".."` is that `version("..")`
-looks like a list.
+However, we have decided to go with the syntax in 2. because the `cfg` flags
+that use the `flag = ".."` syntax are all static as opposed to dynamic.
+In other words, the semantics of `cfg(x = "y")` is equivalent to
+*"`rustc --print=cfg` contains a `x="y"` line"*.
```

Comment:
Maybe this should explicitly quote me, because I was giving a rough approximation here. Maybe someone else in @rust-lang/compiler can chip in? Maybe @petrochenkov?

My point was that `cfg(x)` and `cfg(x = "y")` are "testing for membership" in a fixed set, determined ahead of time, that *happens* to also be available through `rustc --print=cfg`.

But if `cfg(version(...))` does a dynamic semver check, then that's *fundamentally a different operation*, and the same reasoning as for `cfg(accessible(...))` applies.

If I saw `cfg(version = "1.29.0")`, my *instinct* would be that it's an *exact* version check, because that's how *literally* every other instance of `cfg(x = "y")` works, whether built-in or supplied through `--cfg`.